src/CMake/VisItMacros.cmake

```diff
@@ -263,3 +263,72 @@ function(ADD_CMAKE_GEN_TARGET gen_name
 endfunction()
 
 
+##############################################################################
+# This macro appends to a CACHE var list denoted by the NAME argument.
+# Caller is responsible for unsetting the CACHE var when no longer needed
+# to avoid polluting the CACHE.
+#
+# Designed mainly for targets whose sources live in subdirectories,
+# as a means for them to add to the parent's list of source/includes/etc
+# to prepare for a 'blt_add_library' call which requires SOURCES.
+#
+##############################################################################
+
+macro(visit_append_list)
+    cmake_parse_arguments(arg "" "NAME" "ITEMS" ${ARGN})
+    if(NOT DEFINED arg_NAME OR NOT DEFINED arg_ITEMS)
+        message(FATAL_ERROR "visit_append_list called with invalid arguments. Must supply 'NAME' (name of list) and 'ITEMS' (list of items to be added to named list.)")
+    endif()
+    set(${arg_NAME} ${${arg_NAME}} ${arg_ITEMS} CACHE STRING "" FORCE)
+endmacro()
+
+##############################################################################
+# Adds a library target. Wrapper around blt_add_library so that CACHE vars
+# possibly created by visit_append_list for forming SOURCES/INCLUDES, etc
+# can be unset.
+#
+# ARGUMENTS:
+#    NAME         library name               REQUIRED
+#    SOURCES      [source1 [source2 ...]]    REQUIRED
+#    INCLUDES     [dir1 [dir2 ...]]          OPTIONAL
+#    DEFINES      [define1 [define2 ...]]    OPTIONAL
+#    DEPENDS      [dep1 ...]                 OPTIONAL
+#    OUTPUT_NAME  [name]                     OPTIONAL
+#    FOLDER       [name])                    OPTIONAL
+#
+##############################################################################
+
+macro(visit_add_library)
+    set(options)
+    set(singleValueArgs NAME OUTPUT_NAME FOLDER)
+    set(multiValueArgs SOURCES INCLUDES DEFINES DEPENDS)
+
+    # parse the arguments
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    # Sanity checks
+    if(NOT arg_NAME)
+        message(FATAL_ERROR "visit_add_library() must be called with argument NAME <name>")
+    endif()
+
+    if (NOT arg_SOURCES)
+        message(FATAL_ERROR "visit_add_library(NAME ${arg_NAME} ...) called with no given sources")
+    endif()
+
+    blt_add_library(
+        NAME       ${arg_NAME}
+        SOURCES    ${arg_SOURCES}
+        INCLUDES   ${arg_INCLUDES}
+        DEFINES    ${arg_DEFINES}
+        DEPENDS_ON ${arg_DEPENDS}
+        FOLDER     ${arg_FOLDER})
+
+    # vars that may have been created by calls to visit_append_list
+    unset(${arg_NAME}_SOURCES CACHE)
+    unset(${arg_NAME}_INCLUDES CACHE)
+    unset(${arg_NAME}_DEFINES CACHE)
+    unset(${arg_NAME}_DEPENDS CACHE)
+    unset(${arg_NAME}_FEATURES CACHE)
```

Comment:
So, I see `${arg_NAME}_FEATURES` here but it is not used in `blt_add_library()`

Reply:
It is not an option for blt_add_library.  We have to call target_compile_features.  I just made a commit that makes use of FEATURES.